src/init.c

```diff
@@ -1535,6 +1537,35 @@ cmd_spec_header (const char *com, const char *val, void *place_ignored _GL_UNUSE
   return true;
 }
 
+static bool
+cmd_spec_resolve_host (const char *com _GL_UNUSED, const char *val, void *place_ignored _GL_UNUSED)
+{
+  //parse host:ip and put into dns cache
+
+  char* pos = strpbrk(val, ":");
+  bool result = false;
+  if (pos && pos != val && pos != val + strnlen(val, 256))
+  {
+    size_t len = strnlen(val, 256);
```

Comment:
I would avoid code duplication and reduce 2N to N by moving `len` variable out of the cycle and reusing it.

Reply:
@mutanabbi I frankly do no get your idea, as there's no loop there. Did you miss the `if` with cycle? It looks very similar.